packages/cmd/src/CommandProcessor.ts

```diff
@@ -707,19 +708,112 @@ export class CommandProcessor {
+            }
+            fromCnfg = { ...fromCnfg, ...p };
+        }
+
+        // Convert each property extracted from the config to the correct yargs
+        // style cases for the command handler (kebab and camel)
+        allOpts.forEach((opt) => {
+            const cases = CliUtils.getOptionFormat(opt.name);
+            const profileKebab = fromCnfg[cases.kebabCase];
+            const profileCamel = fromCnfg[cases.camelCase];
+
+            if ((profileCamel !== undefined || profileKebab !== undefined) &&
+                (!args.hasOwnProperty(cases.kebabCase) && !args.hasOwnProperty(cases.camelCase))) {
+
+                // If both case properties are present in the profile, use the one that matches
+                // the option name explicitly
+                const value = (profileKebab !== undefined && profileCamel !== undefined) ?
+                    ((opt.name === cases.kebabCase) ? profileKebab : profileCamel) :
+                    ((profileKebab !== undefined) ? profileKebab : profileCamel);
+                const keys = CliUtils.setOptionValue(opt.name,
+                    ("aliases" in opt) ? (opt as ICommandOptionDefinition).aliases : [],
+                    value
+                );
+                fromCnfg = { ...fromCnfg, ...keys };
+            }
+        });
+
+        // Merge the arguments from the config into the CLI args
+        this.log.trace(`Arguments extract from the config:\n${inspect(fromCnfg)}`);
+        args = CliUtils.mergeArguments(fromCnfg, args);
+
         // Load all profiles for the command
         this.log.trace(`Loading profiles for "${this.definition.name}" command. ` +
-            `Profile definitions: ${inspect(this.definition.profile, {depth: null})}`);
+            `Profile definitions: ${inspect(this.definition.profile, { depth: null })}`);
 
         const profiles = await CommandProfileLoader.loader({
             commandDefinition: this.definition,
             profileManagerFactory: this.profileFactory
         }).loadProfiles(args);
-        this.log.trace(`Profiles loaded for "${this.definition.name}" command:\n${inspect(profiles, {depth: null})}`);
+        this.log.trace(`Profiles loaded for "${this.definition.name}" command:\n${inspect(profiles, { depth: null })}`);
 
         // If we have profiles listed on the command definition (the would be loaded already)
         // we can extract values from them for options arguments
         if (this.definition.profile != null) {
+
+            // "fake out" the cli util to only populate options for profiles
+            // that have not been fulfilled by the config
+            const p: ICommandProfile = {
+                required: [],
+                optional: [],
+                suppressOptions: this.definition.profile.suppressOptions
+            };
+
+            if (this.definition.profile.required)
+                p.required = this.definition.profile.required.filter(type => fulfilled.indexOf(type) < 0);
```

Comment:
I may be missing something, but it looks like p.required and p.optional are set, but never used.

Reply:
Good catch, thank you. The "faked" definition is supposed to be passed to `CliUtils.getOptValueFromProfiles` to avoid including options from profiles that are already "fulfilled" by the config. I'll update in my next commit. 

I'll defer to my PR disclaimer `NOTE: Not everything may function exactly as described... let me know if you have issues. Tests have yet to be written.` 😅